tests/test_common_view_logic.py

```diff
@@ -1,389 +1,182 @@
-        get_group_by_id.return_value = "test_group"
-        articles = [
-            {
-                "featured_media": "test",
-                "author": "test",
-                "categories": [1, 2],
-                "group": [1],
-                "tags": ["test"],
-            },
-            {
-                "featured_media": "test2",
-                "author": "test2",
-                "categories": [2, 3],
-                "group": [1],
-                "tags": ["test2"],
-            },
-        ]
-        context = get_index_context(1, articles, 2)
-        expected_context = {
-            "current_page": 1,
-            "total_pages": 2,
-            "featured_articles": [],
-            "articles": [
-                {
-                    "author": "test_author",
-                    "categories": [1, 2],
-                    "display_category": "test_category",
-                    "featured_media": "test",
-                    "group": "test_group",
-                    "image": "test_image",
-                    "tags": ["test"],
-                },
-                {
-                    "author": "test_author",
-                    "categories": [2, 3],
-                    "display_category": "test_category",
-                    "featured_media": "test2",
-                    "group": "test_group",
-                    "image": "test_image",
-                    "tags": ["test2"],
-                },
-            ],
-            "groups": {1: "test_group"},
-            "upcoming": [],
-            "used_categories": {
-                1: "test_category",
-                2: "test_category",
-                3: "test_category",
-            },
-        }
-        self.maxDiff = None
-        self.assertEqual(context, expected_context)
-
-    @patch("canonicalwebteam.blog.wordpress_api.get_group_by_id")
-    @patch("canonicalwebteam.blog.wordpress_api.get_category_by_id")
-    @patch("canonicalwebteam.blog.wordpress_api.get_user")
-    @patch("canonicalwebteam.blog.wordpress_api.get_media")
-    def test_building_index_context_with_featured_articles(
-        self, get_media, get_user, get_category_by_id, get_group_by_id
-    ):
-        get_media.return_value = "test_image"
-        get_user.return_value = "test_author"
-        get_category_by_id.return_value = "test_category"
-        get_group_by_id.return_value = "test_group"
-        articles = [
-            {
-                "featured_media": "test",
-                "author": "test",
-                "categories": [1, 2],
-                "group": [1],
-                "tags": ["test"],
-            },
-            {
-                "featured_media": "test2",
-                "author": "test2",
-                "categories": [2, 3],
-                "group": [1],
-                "tags": ["test2"],
-            },
+    @vcr.use_cassette("fixtures/vcr_cassettes/get_complete_article.yaml")
+    def test_get_complete_article(self):
```

Comment:
I feel here you are just testing that your fixtures are well written/

If you want to have exhaustive you should try to break your application instead:

- What happens if you give an empty object?
- What happens if there is no catogory/group/author?

Each endpoint IMO should be in it's own class and file, even if in the meantime there will be only one function per class. Otherwise you might end with a very very long test suite file.

Reply:
The fixtures are all auto generated from the API call via https://vcrpy.readthedocs.io/en/latest/ .

But yes, the test relies on the library correctly creating the fixture.

I think breaking the app is a good idea. Ill try to add some of those.

Reply:
https://github.com/canonical-web-and-design/canonicalwebteam.blog/issues/70 

Created this to tackle the failing tests